### ISSUE TITLE
xcms integration: chromatogram deconvolution fails with recent version of xcms

### ISSUE THREAD

**Issue by rababerladuseladim** [mzmine/mzmine2#298]
Windows Version: 10 x64
java: v8 update 121
R: 3.4.0
mzmine: 2.24
xcms: 1.52.0

Hi, 
we recently updated our R installation and since the update we cannot deconvolute our chromatograms with the wavelet mthod of xcms anymore.

The associated error message:

`[16:16:27|FINEST|MainMenu]: Setting parameters for module Chromatogram deconvolution
[16:16:49|FINEST|PeakResolverSetupDialog]: Loading new preview peak #1 84.960 m/z @20.5
[16:16:49|FINEST|RLocationDetection]: Windows: Query registry to find where R is installed ...
[16:16:49|FINEST|RLocationDetection]: R InstallPath = C:\Program Files\R\R-3.4.0
[16:16:49|FINEST|RLocationDetection]: R executable location auto-detected as 'C:\Program Files\R\R-3.4.0\bin\R.exe'
[16:16:50|FINEST|RSessionWrapper]:   failed: timed out
[16:16:50|FINEST|RSessionWrapper]: Connection R://MZmineUser:8d4e3598-6651-406f-b430-2c9689042819@localhost:6311 failed.
[16:16:50|FINEST|RSessionWrapper]: !! Rserve R://MZmineUser:8d4e3598-6651-406f-b430-2c9689042819@localhost:6311 is not accessible.
 ! null
[16:16:50|FINEST|RSessionWrapper]: Trying to spawn R://MZmineUser:8d4e3598-6651-406f-b430-2c9689042819@localhost:6311
[16:16:50|FINEST|RSessionWrapper]: Failed to connect: Cannot connect: Connection refused: connect
[16:16:50|FINEST|RLocationDetection]: Windows: Query registry to find where R is installed ...
[16:16:50|FINEST|RLocationDetection]: R InstallPath = C:\Program Files\R\R-3.4.0
[16:16:50|FINEST|RLocationDetection]: R executable location auto-detected as 'C:\Program Files\R\R-3.4.0\bin\R.exe'
[16:16:50|FINEST|RSessionWrapper]: checking Rserve is available... 
[16:16:50|FINEST|RSessionWrapper]:   executing "C:\Program Files\R\R-3.4.0\bin\R.exe" -e "i=installed.packages();is.element(set=i,el='Rserve')" --vanilla -q
[16:16:50|FINEST|RSessionWrapper]:   ok
[16:16:50|FINEST|RSessionWrapper]: starting R daemon... R://MZmineUser:8d4e3598-6651-406f-b430-2c9689042819@localhost:6311
[16:16:50|FINEST|RSessionWrapper]: Waiting for Rserve to start ...
[16:16:50|FINEST|RSessionWrapper]:   executing "C:\Program Files\R\R-3.4.0\bin\R.exe" -e "library(Rserve);Rserve(FALSE,args='--no-save --slave  --RS-pidfile "C:/Users/L7CA9~1.KNO/AppData/Local/Temp/MZmine2017-05-02_16-11-21.719/rs_pid_63114551481648269432115.pid" --RS-port 6311')" --no-save --slave
[16:16:50|FINEST|RSessionWrapper]: Rserve startup done, let us try to connect ...
[16:16:50|FINEST|RSessionWrapper]: Rserve is running.
[16:16:50|FINEST|RSessionWrapper]:   ok
[16:16:51|FINEST|RSessionWrapper]: Local Rserve started. (Version 103)
[16:16:51|FINEST|RSessionWrapper]: Rserve: started instance (pid: '8828' | port: '6311').
[16:16:51|FINEST|RSessionWrapper]: R version 3.4.0 (2017-04-21)
[16:16:51|FINEST|RSessionWrapper]: 103
[16:16:51|FINEST|RSessionWrapper]: UTF-8
[16:16:51|FINEST|RSessionWrapper]: Loading package 'xcms'...
[16:16:56|FINEST|RSessionWrapper]: Load status: 'true'.
[16:16:56|FINEST|RSessionWrapper]: Loaded package: 'xcms'.
[16:16:56|FINEST|RSessionWrapper]: Checking package version: 'xcms' for version '1.33.2'...
[16:16:56|FINEST|RSessionWrapper]: Checked package version: 'xcms' for version '1.33.2'.
[16:16:56|FINEST|CentWaveDetector]: Detecting peaks.
[16:16:56|FINEST|RSessionWrapper]: !     Repeated 13 times.
[16:16:56|FINEST|RSessionWrapper]: ! [exception] Error in .centWave_orig(mz = mz, int = int, scantime = scantime, valsPerSpect = valsPerSpect,  : 
  Lengths of 'mz', 'int' and of 'scantime','valsPerSpect' have to match. Also, 'length(mz)' should be equal to 'sum(valsPerSpect)'.

  findPeaks.centWave(xRaw, ppm=0, mzdiff=0, verbose=TRUE, peakwidth=c(300.0, 600.0), snthresh=10.0, integrate=1, ROI.list=ROIs)
[16:16:56|FINEST|RSessionWrapper]: Rserve: try terminate  session (pid: '8828' | port: '6311')...
[16:16:56|FINEST|RSessionWrapper]: Ending remote session...
[16:16:56|FINEST|RSessionWrapper]: Session teminated.
[16:16:56|FINEST|RSessionWrapper]: Rserve: terminated  session (pid: '8828' | port: '6311')...`

With the old installation (3.3.2) of R and the respective xcms version (1.50.1) everything works as expected.

**Comment by MattyRoth90** [comment]
Hi,

I confirm the same problem with R 3.4.0

**Comment by Ateabening** [comment]
Same with me. 
It could not perform the deconvolution in wavelet. 

message:
[12:01:52|SEVERE|WorkerThread]: Error of task Peak recognition on BM-D10-1-pos.raw baseline-corrected chromatograms: 'R computing error' during CentWave detection.
The "Wavelets (XCMS)" requires the "xcms" R package, which couldn't be loaded - is it installed in R?
Please fix.

**Comment by rababerladuseladim** [comment]
@Ateabening: you need to install the packages mentioned und er "prerequisites".
http://mzmine.github.io/download.html

**Comment by Ateabening** [comment]
I already installed the prerequisites but it didn't work.

**Comment by rababerladuseladim** [comment]
@Ateabening:
Your error message states that xcms could not be loaded and therefore probably is not installed. Whatever the reason, you are facing a different error than the one this issues was opened about.

Did you check within R if the package is listed as installed?
Do you have multiple R installations on your PC and mzMine is loading an installation where xcms is not installed?

**Comment by Ateabening** [comment]
I have installed  R x 64 3.1.2 and R i386 3.13  and R studio in my
computer.

Asih Triastuti
PhD Student
Pharmacochimie et Pharmacologie Pour le Développement (PHARMA-DEV )
IRD UMR 152
35 chemin des Maraîchers,
31400 TOULOUSE
FRANCE

**Comment by rababerladuseladim** [comment]
@Ateabening 
What you need to do (in a R console) is execute these lines:
```
install.packages(c("Rserve", "ptw", "gplots", "baseline", "hyperSpec", "ggplot2"))
source("http://bioconductor.org/biocLite.R")
biocLite("xcms")
biocLite("CAMERA")
biocLite("PROcess")
```
As i do not know which version of R is loaded by mzmine, you should execute these lines once in the R x64 and in the i386 console. These commands are listed as necessary under prerequisites in the mzmine download section.

**Comment by Ateabening** [comment]
I did not install package in my R (both R 64 and i386) because when I
checked them, they were already installed. But then I found solution. I did
the chromatogram deconvolution one by one instead of simultaneously of 71
files and it works!
little bit annoying because you have to wait each 3 minutes before you
click for another file to be done.



Asih Triastuti
PhD Student
Pharmacochimie et Pharmacologie Pour le Développement (PHARMA-DEV )
IRD UMR 152
35 chemin des Maraîchers,
31400 TOULOUSE
FRANCE

**Comment by tomas-pluskal** [comment]
Hi all,
Honestly, nobody is maintaining the R support code in MZmine, so I am not surprised it broke with a new version of R. The code was originally written by @golgauth , but I doubt he has time to update it. I would suggest to try using the ADAP deconvolution module instead, it is based on similar principles like the centWave method in XCMS. I'm sure @du-lab will appreciate any feedback to the ADAP algorithm.

**Comment by golgauth** [comment]
Theoretically fixed here: #321. Should be compatible with older R releases < 3.4.X.

**Comment by tomas-pluskal** [comment]
Confirmed fixed
